google-cloud-datastore/acceptance/datastore/datastore_test.rb

```diff
@@ -424,6 +430,47 @@
       _(entities.count).must_equal 0
     end
 
+    it "should limit results when limit > 300" do
+      kind_val = "Post_#{SecureRandom.hex(4)}"
+      limit = 700
+
+      # Add 1000 entities of the same kind to the datastore
+      1000.times.each do |id|
+        post_temp = post.dup
+        post_temp.key = Google::Cloud::Datastore::Key.new kind_val, "Post_#{id+1}"
+        dataset.save post_temp
+      end
+
+      # Testing limit with query
+      query = dataset.query(kind_val).limit(limit)
+      entities_count = 0
+      results = dataset.run query
+      loop do
+        entities_count += results.count
+        break unless results.next?
+        results = results.next
+      end
+      _(entities_count).must_equal limit
+
+      # Testing limit with GQL query
```

Comment:
Can we split this into a separate test ?
It is generally a good practice to assert for one scenario in unit test, will help pin pointing the issue easier in future and also help understanding the functionality.

Reply:
Splitted the test into 2 parts.